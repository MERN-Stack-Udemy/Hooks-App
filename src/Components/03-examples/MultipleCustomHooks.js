import React from 'react'

export const MultipleCustomHooks = () => {
  return (
    <div>
      <h3>MultipleCustomHooks</h3>
    </div>
  )
}
