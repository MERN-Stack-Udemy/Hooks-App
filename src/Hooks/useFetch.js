import { useEffect, useRef, useState } from 'react'

export const useFetch = ( url ) => {
 
  const [state, setState] = useState({
    data: null,
    error: null,
    loading: true
  });  

  const isMounted = useRef(true);

  useEffect(() => {
    return () => {
      isMounted.current = false;
    }
  }, [])

  useEffect(() => {

    setState({ data:null, error:null, loading:true })
    if(isMounted){

      fetch ( url )
        .then( resp => resp.json())
        .then( data => {
  
          if(isMounted.current){
            setState({
              data,
              error: null,
              loading: false
            });
          }
      
        })
        .catch( () => {
          setState({
            data:null,
            loading: false,
            error: 'request could not be fulfilled',
          })
        })
    }
  }, [url]);
  
  return state;
}
