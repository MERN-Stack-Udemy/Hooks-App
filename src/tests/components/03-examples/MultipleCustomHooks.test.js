import { shallow } from 'enzyme'
import { MultipleCustomHooks } from '../../../Components/03-examples/MultipleCustomHooks'
import { useCounter } from '../../../Hooks/useCounter';
import { useFetch } from '../../../Hooks/useFetch';

jest.mock('../../../Hooks/useFetch');
jest.mock('../../../Hooks/useCounter');


describe('test on MultipleCustomHooks', () => {

  beforeEach( () => {
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {},
      decrement: () => {}
    });
  })


  test('should to match snapshot', () => {
    useFetch.mockReturnValue({
      data: null,
      loading:true,
      error:null
    })

    const wrapper = shallow( <MultipleCustomHooks /> );
    expect( wrapper ).toMatchSnapshot();
  })

  test('should display the information', () => {
    
    useFetch.mockReturnValue({
      data: [{
        author: 'Abdiel',
        quote: 'hola mundo'
      }],
      loading:false,
      error:null
    })
    const wrapper = shallow( <MultipleCustomHooks/> );

    expect( wrapper.find('.alert').exists() ).toBe(false);
    expect( wrapper.find('.mb-2').text().trim() ).toBe( 'hola mundo' );
    expect( wrapper.find('footer').text().trim() ).toBe( 'Abdiel' );

  })
  
  
})
