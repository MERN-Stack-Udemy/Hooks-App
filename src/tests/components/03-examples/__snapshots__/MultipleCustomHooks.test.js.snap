// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on MultipleCustomHooks should to match snapshot 1`] = `
<div>
  <h3>
    Breaking Bad Quotes 
  </h3>
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary m-1"
    onClick={[Function]}
  >
    Previus Quote
  </button>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
